autogluon/task/tabular_prediction/tabular_prediction.py

```diff
@@ -229,7 +229,14 @@ def fit(train_data, label, tuning_data=None, output_directory=None, problem_type
         id_columns = kwargs.get('id_columns', [])
         trainer_type = kwargs.get('trainer_type', AutoTrainer)
         nthreads_per_trial, ngpus_per_trial = setup_compute(nthreads_per_trial, ngpus_per_trial)
-        time_limits, num_trials = setup_trial_limits(time_limits, num_trials, hyperparameters)
+        time_limits_orig = copy.deepcopy(time_limits)
+        time_limits_hpo = copy.deepcopy(time_limits)
+        if num_bagging_folds >= 2:
+            time_limits_hpo = time_limits_hpo / (1 + num_bagging_folds * (1 + stack_ensemble_levels))
+        time_limits_hpo, num_trials = setup_trial_limits(time_limits_hpo, num_trials, hyperparameters)  # TODO: Move HPO time allocation to Trainer
+        if (num_trials is not None) and hyperparameter_tune and (num_trials == 1):
+            hyperparameter_tune = False
+            logger.log(30, 'Warning: Specified time_limits is too small for hyperparameter_tune, setting to False.')
```

Comment:
Isn't num_trials what is specified too small here?

Reply:
True but this can also occur because time_limits was small and thus num_trials was set to 1. O agree it is a bit confusing

Reply:
Updated